Fix: remove spaces from reply to MXP version tag.

eowmob opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Removes spaces from the version string returned by MXP to avoid parsing issues in the mud. Also, like other MXP clients, mudlet will now report the version nummer without prefixing the client name (which is reported in a separate field).
#### Motivation for adding to Mudlet
Improve MXP compatibility
#### Other info (issues closed, discussion etc)
This is meant as a fix to issue #7140 which popped up in recent PTB versions when the format of the internal version string was changed.

eowmob committed:
Fixed malformed version tag

eowmob committed:
Improved comments a bit

vadi2 commented:
Thanks for throwing a PR together. Could you also test https://github.com/Mudlet/Mudlet/pull/7138 ?

eowmob committed:
Merge branch 'development' into MXPVersionWithoutSpace

eowmob commented:
> Could you also test https://github.com/Mudlet/Mudlet/pull/7138 ?

Sure, I had already resynched my PR to the change of the development tree, so the test versions above contain the fix to both #7138 and my fix to #7140 (they really do, though the bot msg is shown before my merge entry, but check the build hash) . My PR and #7138 address somewhat different issues: I did not notice the GMCP issue (sorry), I noticed MXP parsing (in the mud) has issues when the version number was now prefixed with "mudlet " rather than "mudlet-".

As I already resynched or rebased my PR to the current development, I can say both fixes are compatible and deal with separate issues (note I did a bunch of recompiles, test version downloads etc. to check my fix and #7138 fix the resp. MXP / GMCP issue alone, nothing more). I was wondering though, why the trailing \n in the version string did not manifest in MXP.

After some gdb debugging I'm confident this boils down to ctelnet.cpp:607 bool cTelnet::sendData() which probably removes the additional \n with data.remove(QChar::LineFeed); in line 609 . So the \n at the end of the version string in the reply to &lt;VERSION> never makes it to the mud.

eowmob committed:
Clarified comment

explain we want to avoid having a space and the client name twice in the VERSION attribute of the <VERSION> tag.

eowmob committed:
Merge branch 'Mudlet:development' into MXPVersionWithoutSpace

eowmob committed:
Merge branch 'Mudlet:development' into MXPVersionWithoutSpace
